tests/geo/test_find_in_string.py

```diff
@@ -0,0 +1,481 @@
+        When the input string is distorted too much it should return an empty list
+        """
+        test_samples = [
+            (
+                "Department of Biological Oceanography, Royal Institute for Sea Research (NIOZ), Texel",
+                [],
+            ),
+            (
+                "International Centre for Radio Astronomy Research (ICRAR), M468, University of West Crawley, WA 6009",
+                [],
+            ),
+            (
+                "Institute of Biochemistry and Genetics, Ufa Science Center, Academy of Sciences, Ufa",
+                [],
+            ),
+        ]
+        for original, expected_match in test_samples:
+            with self.subTest(original=original, expected_match=expected_match):
+                extracted_country = find_country_in_string(original)
+                self.assertEqual(extracted_country, expected_match)
+
+
+class FindCityInStringTest(unittest.TestCase):
+    def test_find_multiple_cities_in_string(self):
+        """
+        When the input string is distorted too much it should return an empty list
+        """
+        test_samples = [
+            (
+                "Météorage Pau France",
+                {"France"},
+                [
+                    Match(
+                        substring_range=(10, 13),
+                        substring="Pau",
+                        normalized=("Pau", "France"),
+                        score=1.0,
+                    ),
+                    Match(
+                        substring_range=(14, 20),
+                        substring="France",
+                        normalized=("La Frasnée", "France"),
+                        score=0.9090909090909091,
+                    )
+                ],
+            ),
+            (
+                "Bavarian Environment Agency, Hans Högn Straße 12, 95030 Hof Saale, Bavaria, Germany",
+                {"Germany"},
+                [
+                    Match(
+                        substring_range=(56, 59),
+                        substring="Hof",
+                        normalized=("Hof", "Germany"),
+                        score=1.0,
+                    ),
+                    Match(
+                        substring_range=(39, 45),
+                        substring="Straße",
+                        normalized=("Trassem", "Germany"),
+                        score=0.8571428571428572,
+                    )
+                ],
+            ),
+            (
+                "Institute of Molecular Medicine and Max Planck Research Department of Stem Cell Aging, University of "
+                "Ulm, Ulm, Germany",
+                {"Germany"},
+                [
+                    Match(
+                        substring_range=(101, 104),
+                        substring='Ulm',
+                        normalized=('Ulm', 'Germany'),
+                        score=1.0
+                    ),
+                    Match(substring_range=(106, 109),
```

Comment:
Your consistency is a bit off here and there. In previous lines you used a newline right after `Match(`